src/axom/core/ArrayBase.hpp

```diff
@@ -812,7 +812,7 @@ struct ArrayOpsBase<T, false>
                          const T* values,
                          MemorySpace space)
   {
-#if defined(AXOM_GPUCC) && defined(AXOM_USE_UMPIRE)
+#if defined(AXOM_USE_GPU) && defined(AXOM_GPUCC) && defined(AXOM_USE_UMPIRE)
```

Comment:
To get this to work, Alan replaced `defined(AXOM_GPUCC)` with `defined(AXOM_USE_CUDA)`.  I changed this by adding (not replacing with) the more general test on AXOM_USE_GPU.

Reply:
Thanks @agcapps! 
For reference:

- `AXOM_USE_GPU` is defined when Axom is configured through CMake to build with HIP or CUDA (`ENABLE_HIP` / `ENABLE_CUDA`)
- `AXOM_GPUCC` is defined where code is compiled by hip clang (amdclang) or nvcc compilers

I think `AXOM_GPUCC` used to matter when not all parts of the code were compiled by amdclang/nvcc when Axom was configured with CUDA/HIP.

In order to allow Axom to build on newer CMake versions with device (PR #990), I think all Axom code now compile with amdclang/nvcc if Axom was configured with CUDA/HIP.

Should we deprecate use of `AXOM_GPUCC` in a separate PR?

Reply:
Got it.  Let us consider getting rid of `AXOM_GPUCC` at some point.

Reply:
@agcapps @bmhan12 will one of you please make an issue to resolve this so we don't forget? Thank you.

Reply:
Done: see issue #1101

Reply:
See related conversation below.